### DISCUSSION TITLE
Extra 'v' in newValue

### DISCUSSION THREAD

**Discussion by o-voroniak** [renovatebot/renovate#22638] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.113.2

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Hi, folks.
I just faced a strange issue with the dependency update in Podfle. 
To reproduce, create an empty repo with Podfile in it (you can skip Podfile.lock):

```
target 'TestAPP' do   
  pod 'NeedleFoundation', :git => 'https://github.com/uber/needle.git', :tag => 'v0.22.0'   
end   
```

Run renovate, and it will create an MR with updating to the vv23, instead of v23:

`pod '[NeedleFoundation](https://github.com/uber/needle.git)', :git => 'https://github.com/uber/needle.git', :tag => '**vv0.23.0**'`

The strange thing is that in MR's description, everything is fine - there is no extra V:
`Update dependency NeedleFoundation to v0.23.0`

Commit name is also acceptable:
`Update dependency NeedleFoundation to v0.23.0`

In the logs, we can see that newVersion and newValue are different:
`newVersion": "v0.23.0",
"newValue": "vv0.23.0",`

I'm struggling to understand why it works like this. Any suggestions?

### Relevant debug logs

<details><summary>Logs</summary>

"config": {
         "cocoapods": [
           {
             "deps": [
               {
                 "datasource": "github-tags",
                 "depName": "NeedleFoundation",
                 "packageName": "uber/needle",
                 "currentValue": "v0.22.0",
                 "managerData": {"lineNumber": 1},
                 "updates": [
                   {
                     "bucket": "non-major",
                     **"newVersion": "v0.23.0",
                     "newValue": "vv0.23.0",**
                     "releaseTimestamp": "2023-04-27T21:55:14.000Z",
                     "newMajor": 0,
                     "newMinor": 23,
                     "updateType": "minor",
                     "isRange": true,
                     "branchName": "renovate-needlefoundation-0.x"
                   }
                 ],
                 "warnings": [],
                 "versioning": "ruby",
                 "sourceUrl": "https://github.com/uber/needle",
                 "registryUrl": "https://github.com/",
                 "currentVersion": "v0.22.0",
                 "isSingleVersion": false,
                 "fixedVersion": "v0.22.0"
               }
             ],
             "packageFile": "Podfile"
           }
         ]
       }

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by MindTooth** [comment]
Using `versioning = semver` "fixed" it: https://github.com/MindTooth/renovate-cocoapods-test/pull/2

```json
  "packageRules": [
    {
      "matchPackageNames": ["NeedleFoundation"],
      "versioning": "semver"
    }
  ]
  ```
  
  Guess it must be something about the Ruby versioning that adds the extra `v`? 🤷🏻‍♂️ 
  
https://github.com/renovatebot/renovate/blob/1d3687e80d5e0a1f52f0d8f7582bfe02eb0589c3/lib/modules/versioning/ruby/index.ts#L106

~~I might be reading this wrong, but does it not say that if it starts with a `v`, add another one?~~  Guess `newVersion` is without `v`.

**Reply by viceice** [reply]
we've probably a bug inside ruby versioning. we should not add a v prefix if already there.
